File: Go/Programmers/big_number_behind.go
package programmers

func bigNumberBehind(numbers []int) []int {
	answer := make([]int, len(numbers))
	big_nums := []int{}
	for i := len(numbers) - 1; i > -1; i-- {
		for j := len(big_nums) - 1; j > -1; j-- {
			if numbers[i] < big_nums[j] {
				answer[i] = big_nums[j]
				big_nums = append(big_nums[0:j+1], numbers[i])
				break
			}
		}
		if answer[i] == 0 {
			answer[i] = -1
			big_nums = []int{numbers[i]}
		}
	}
	return answer
}

File: Java/Programmers/big_number_behind.java
package Programmers;

import java.util.Arrays;
import java.util.Stack;

public class big_number_behind {
    public int[] solution(int[] numbers) {
        int[] answer = new int[numbers.length];
        Arrays.fill(answer, -1);
        Stack<Integer> stack = new Stack<>();
        for (int i = 0; i < numbers.length; i++){
            while (!stack.empty() && numbers[stack.elementAt(stack.size())] < numbers[i]) {
                answer[stack.pop()] = numbers[i];
            }
            stack.push(i);
        }
        return answer;
    }
}
